### DISCUSSION TITLE
X-Frame-Options question

### DISCUSSION THREAD

**Discussion by YouveGotMeowxy** [authelia/authelia#3429] [Q&A] [answered]
I am running most of my containers/services within an iframe in Organizr. If I'm not logged into Authelia with a user, any of the services I have protected with it show this in the console:

`Refused to display 'https://servicename.mydomain.com/' in a frame because it set 'X-Frame-Options' to 'sameorigin'.`

Once I'm logged into Authelia with a user, those same services load in the iframe fine.

Normally, I can just add:

`proxy_hide_header X-Frame-Options;`

to the reverse proxy config for Nginx to get a service to load in the iframe, and I did so with Authelia's reverse proxy, but iit still happens. How can I solve this so that when logged out, the normal Authelia login screens still show in the iframe? (I do understand it removes a layer of security).

**Comment by james-d-elliott** [comment, marked as answer]
> I do understand it removes a layer of security

I think that not understanding the implications completely may be an issue, and for anyone else who decides to do this I need to make it crystal clear the potential for exploitation. What it allows is for an attacker to trick any of your users including yourself into entering details into the wrong site. Then transparent to the user, entering the details into the actual site so they don't even know that they were compromised. 

Authelia doesn't do anything special in this regard, it just adds the X-Frame-Options header. Using proxy_hide_header in nginx is sufficient to remove it. If it's still complaining then there is an issue with how you configured it. Check the headers Authelia is responding with when you visit it directly?
